pkg/components/contour/component_test.go

```diff
@@ -22,41 +22,72 @@ import (
 	"github.com/kinvolk/lokomotive/pkg/components/util"
 )
 
-func testRenderManifest(t *testing.T, configHCL string) {
-	body, diagnostics := util.GetComponentBody(configHCL, name)
-	if diagnostics != nil {
-		t.Fatalf("Error getting component body: %v", diagnostics)
-	}
-
-	component := newComponent()
-	diagnostics = component.LoadConfig(body, &hcl.EvalContext{})
-	if diagnostics.HasErrors() {
-		t.Fatalf("Valid config should not return error, got: %s", diagnostics)
-	}
-
-	m, err := component.RenderManifests()
-	if err != nil {
-		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
-	}
-	if len(m) <= 0 {
-		t.Fatalf("Rendered manifests shouldn't be empty")
-	}
-}
-
-func TestRenderManifestWithIngressHosts(t *testing.T) {
-	configHCL := `
+//nolint:funlen
+func TestRenderManifest(t *testing.T) {
+	tests := []struct {
+		desc    string
+		hcl     string
+		wantErr bool
+	}{
+		{
+			desc: "With ingress hosts",
+			hcl: `
 component "contour" {
   ingress_hosts = ["test.domain.com"]
 }
-`
-	testRenderManifest(t, configHCL)
-}
-
-func TestRenderManifestWithServiceMonitor(t *testing.T) {
-	configHCL := `
+			`,
+		},
+		{
+			desc: "With service monitor",
+			hcl: `
 component "contour" {
   service_monitor = true
 }
-`
-	testRenderManifest(t, configHCL)
+			`,
+		},
+		{
+			desc: "With service type",
+			hcl: `
+component "contour" {
+  service_type = "NodePort"
+}
+			`,
+		},
+		{
+			desc: "Wrong service type",
+			hcl: `
+component "contour" {
+  service_type = "stuff"
+}
+			`,
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		b, d := util.GetComponentBody(tc.hcl, name)
+		if d != nil {
+			t.Fatalf("Error getting component body: %v", d)
+		}
+
+		c := newComponent()
+		d = c.LoadConfig(b, &hcl.EvalContext{})
```

Comment:
nit: Can't you just use `nil` instead of `&hcl.EvalContext{}`? Checking the docs, it [seems nil will do the same](https://pkg.go.dev/github.com/hashicorp/hcl2/gohcl?tab=doc#DecodeBody):

>  The given EvalContext is used to resolve any variables or functions in expressions encountered while decoding. This may be nil to require only constant values, for simple applications that do not support variables or functions.

I think both are fine, though. This is just nit-picking and my personal taste, no big deal if you prefer it this way for some reason.

Reply:
This, too, was copied from an existing component (cluster-autoscaler). Feel free to open an issue to handle other occurrences.